packages/backend/src/nest/ipfs/ipfs.service.ts

```diff
@@ -1,60 +1,168 @@
+      this.logger.error('IPFS creation failed', error)
+      throw new Error('IPFS creation failed')
     }
 
     return this.ipfsInstance
   }
 
+  private async initializeStores(init?: StoreInit): Promise<void> {
+    this.datastore = await this.createDatastore(init?.datastore)
+    this.blockstore = await this.createBlockstore(init?.blockstore)
+  }
+
+  private async createDatastore(init?: DatabaseOptions<string, Uint8Array>): Promise<LevelDatastore> {
+    let datastoreInit: DatabaseOptions<string, Uint8Array> = {
+      keyEncoding: 'utf8',
+      valueEncoding: 'buffer',
+    }
+
+    if (init != null) {
+      datastoreInit = {
+        ...datastoreInit,
+        ...init,
+      }
+    }
+
+    if (datastoreInit.valueEncoding != 'buffer') {
+      throw new Error(`Datastore valueEncoding was set to ${datastoreInit.valueEncoding} but MUST be set to 'buffer'!`)
+    }
+
+    if (datastoreInit.keyEncoding != 'utf8') {
+      throw new Error(`Datastore keyEncoding was set to ${datastoreInit.keyEncoding} but MUST be set to 'utf8'!`)
+    }
+
+    const datastoreLevelDb = new Level<string, Uint8Array>(this.ipfsRepoPath + '/data', datastoreInit)
+    return new LevelDatastore(datastoreLevelDb, datastoreInit)
+  }
+
+  private async createBlockstore(init?: LevelBlockstoreInit): Promise<LevelBlockstore> {
+    let blockstoreInit: LevelBlockstoreInit = {
+      keyEncoding: 'utf8',
+      valueEncoding: 'buffer',
+      createIfMissing: true,
+      errorIfExists: false,
+      version: 1,
+    }
+
+    if (init != null) {
+      blockstoreInit = {
+        ...blockstoreInit,
+        ...init,
+      }
+    }
+
+    if (blockstoreInit.valueEncoding != 'buffer') {
+      throw new Error(
+        `Blockstore valueEncoding was set to ${blockstoreInit.valueEncoding} but MUST be set to 'buffer'!`
+      )
+    }
+
+    if (blockstoreInit.keyEncoding != 'utf8') {
+      throw new Error(`Blockstore keyEncoding was set to ${blockstoreInit.keyEncoding} but MUST be set to 'utf8'!`)
+    }
+
+    const blockstoreLevelDb = new Level<string, Uint8Array>(this.ipfsRepoPath + '/blocks', blockstoreInit)
+    return new LevelBlockstore(blockstoreLevelDb, blockstoreInit)
+  }
+
+  public async start() {
+    this.logger.info(`Starting IPFS Service`)
+    if (!this.ipfsInstance) {
+      throw new Error('IPFS instance does not exist')
+    }
+
+    this.logger.info(`Opening Helia blockstore`)
+    await this.blockstore!.open()
+
+    this.logger.info(`Opening Helia datastore`)
+    await this.datastore!.open()
+
+    this.logger.info(`Starting Helia`)
+    await this.ipfsInstance.start()
+
+    this.started = true
+    this.logger.info(`IPFS Service has started`)
+  }
+
+  public async isStarted() {
+    return this.started
+  }
+
+  public async stop() {
+    this.logger.info('Stopping IPFS')
+    if (!this.ipfsInstance) {
+      throw new Error('IPFS instance does not exist')
+    }
+
+    await this.blockstore?.db.close()
```

Comment:
This is a little weird but there are issues that can arise from not closing the underlying level DB, apparently.